lib/msf/ui/console/module_command_dispatcher.rb

```diff
@@ -138,7 +138,7 @@ def cmd_check(*args)
     return false unless (args = parse_check_opts(args))
 
     mod_with_opts = mod.replicant
-    mod_with_opts.datastore.import_options_from_hash(args[:datastore_options])
+    mod_with_opts.datastore.merge!(args[:datastore_options])
```

Comment:
Same as other locations does this need conditional based on feature flag?

Reply:
Will revert this line :+1:

I'm bringing back the `import_options_from_hash` method on the datastore object just to reduce the surface area of this PR. I'm marking the method as deprecated, and will align these calls to use `merge!` once we're happy that the new datastore can fully replace the existing datastore

Reply:
Flipping this PR back to draft mode, so I can shuffle things around for a bit :+1: